File: core/mathutil.py
"""Tiny shared math helpers used across motion/perception/safety code."""

from __future__ import annotations

import math

TWO_PI = 2.0 * math.pi


def clamp(value: float, lo: float, hi: float) -> float:
    """Clamp value into [lo, hi]."""
    if value < lo:
        return lo
    if value > hi:
        return hi
    return value


def wrap_angle(a: float) -> float:
    """Wrap an angle (radians) into (-pi, pi]."""
    return ((a + math.pi) % TWO_PI) - math.pi

File: browser-sweetie/src/mathutil.js
// Shared geometry helpers (port of Legacy_Sweetie-Bot/core/mathutil.py). Meters + radians throughout.
export const clamp = (v, lo, hi) => (v < lo ? lo : v > hi ? hi : v);

// Wrap an angle to (-π, π]. Robust for arbitrarily large/negative inputs (JS % keeps the sign of the
// dividend, so normalize into [0, 2π) first, then shift).
export function wrapAngle(a) {
  let x = (a + Math.PI) % (2 * Math.PI);
  if (x < 0) x += 2 * Math.PI;
  return x - Math.PI;
}

export const hypot = (dx, dy) => Math.hypot(dx, dy);
